include/grpc++/impl/codegen/status_code_enum.h

```diff
@@ -143,6 +143,11 @@ enum StatusCode {
   /// Unrecoverable data loss or corruption.
   DATA_LOSS = 15,
 
+  // Service was improperly implemented, violated a gRPC API requirement
```

Comment:
Ok, I've now changed this to use INTERNAL since this is also an invariant failure.